From: Colin Ian King <colin.king@canonical.com>
Date: Sat, 17 May 2014 23:33:44 +0100
Subject: IB/mlx4: fix unitialised variable is_mcast
Patch-mainline: v3.16-rc1
Git-commit: bfdfcfee3c9281e9cd28c0b08235aba1762504a6
References: bnc#868011

Commit 297e0dad7206 ("IB/mlx4: Handle Ethernet L2 parameters for IP
based GID addressing") introduced a bug where is_mcast is now no
longer initialized on the non-multicast condition and so it can be
any random value from the stack.  This issue was detected by cppcheck:

    [drivers/infiniband/hw/mlx4/ah.c:103]: (error) Uninitialized
      variable: is_mcast

Simple fix is to initialise is_mcast to zero.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
  bfdfcfee (IB/mlx4: fix unitialised variable is_mcast)
    references
  297e0dad (patches.drivers/0012-IB-mlx4-Handle-Ethernet-L2-parameters-for-IP-based-G.patch: IB/mlx4: Handle Ethernet L2 parameters for IP based GID addressing)
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/infiniband/hw/mlx4/ah.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/mlx4/ah.c
+++ b/drivers/infiniband/hw/mlx4/ah.c
@@ -73,7 +73,7 @@ static struct ib_ah *create_iboe_ah(stru
 {
 	struct mlx4_ib_dev *ibdev = to_mdev(pd->device);
 	struct mlx4_dev *dev = ibdev->dev;
-	int is_mcast;
+	int is_mcast = 0;
 	struct in6_addr in6;
 	u16 vlan_tag;
 
